### ISSUE TITLE
BrowserSync: only check SCSS files

### ISSUE THREAD

**Issue by VincentMarmiesse** [SnowdogApps/magento2-frontools#257]
Hello,

Is it possible to tell to BrowserSync to check only SCSS files? I tried to set the `files` config in `browser-sync.json` but it seems to check also file in `var` or `phtml` ones.

Thanks.

**Comment by Igloczek** [comment]
Nope, there is no way to simply disable it.
But I'm kinda curious, why you would like to do this.

**Comment by VincentMarmiesse** [comment]
Hi,

Because BrowserSync reloads my page even if I only edit a SCSS file. I didn't have this behaviour on previous versions. Here are the logs, I have only edited the file `app/design/frontend/Project/default/web/css/source/components/_homepage.scss`.

```
[16:37:02] Change detected. Theme: project_default File: app/design/frontend/Project/default/web/css/source/components/_homepage.scss___jb_tmp___
[16:37:02] Resolving inheritance. Theme: project_default
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/Algolia_AlgoliaSearch/templates/autocomplete/category.phtml
[Browsersync] Reloading Browsers...
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/Algolia_AlgoliaSearch/templates/autocomplete/menu.phtml
[Browsersync] Reloading Browsers...
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/Algolia_AlgoliaSearch/templates/autocomplete/product.phtml
[Browsersync] Reloading Browsers...
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/Algolia_AlgoliaSearch/templates/autocomplete.phtml
[Browsersync] Reloading Browsers...
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/Algolia_AlgoliaSearch/templates/instant/wrapper.phtml
[Browsersync] Reloading Browsers...
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/Algolia_AlgoliaSearch/web/internals/common.js
[Browsersync] Reloading Browsers...
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/Magento_Catalog/layout/catalog_category_view.xml
[Browsersync] Reloading Browsers...
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/Magento_Catalog/templates/category/products.phtml
[Browsersync] Reloading Browsers...
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/Magento_Catalog/templates/product/list.phtml
[Browsersync] Reloading Browsers...
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/Magento_Catalog/templates/product/view/details.phtml
[Browsersync] Reloading Browsers...
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/Magento_Catalog/templates/product/view/opengraph/general.phtml
[Browsersync] Reloading Browsers...
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/Magento_Email/email/footer.html
[Browsersync] Reloading Browsers...
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/Magento_LayeredNavigation/templates/layer/state.phtml
[Browsersync] Reloading Browsers...
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/Magento_LayeredNavigation/templates/layer/view.phtml
[Browsersync] Reloading Browsers...
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/Magento_Theme/templates/html/breadcrumbs.phtml
[Browsersync] Reloading Browsers...
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/Magento_Theme/templates/messages.phtml
[Browsersync] Reloading Browsers...
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/Magento_Theme/templates/page/js/require_js.phtml
[Browsersync] Reloading Browsers...
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/Magento_Theme/web/favicon.ico
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/Magento_Theme/web/js/view/messages.js
[Browsersync] Reloading Browsers...
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/etc/view.xml
[Browsersync] Reloading Browsers...
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/i18n/fr_FR.csv
[Browsersync] Reloading Browsers...
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/media/preview.jpg
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/css/main.css
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/css/main.scss
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/css/source/_global.scss
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/css/source/_layout.scss
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/css/source/_typography.scss
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/css/source/_vars.scss
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/css/source/blank/_styles-l.scss
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/css/source/blank/_styles-m.scss
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/css/source/components/_algolia.scss
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/css/source/components/_appointment.scss
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/css/source/components/_blog.scss
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/css/source/components/_catalogrequest.scss
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/css/source/components/_catalogsearch.scss
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/css/source/components/_categories.scss
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/css/source/components/_configurator.scss
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/css/source/components/_contact.scss
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/css/source/components/_findapro.scss
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/css/source/components/_footer.scss
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/css/source/components/_header.scss
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/css/source/components/_homepage.scss
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/css/source/components/_maps.scss
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/css/source/components/_message.scss
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/css/source/components/_news.scss
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/css/source/components/_page.scss
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/css/source/components/_product.scss
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/css/source/components/_quality.scss
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/css/source/components/_service.scss
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/css/source/components/_shop.scss
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/css/source/components/_sitemap.scss
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/css/source/components/_slick.scss
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/css/source/components/_terms.scss
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/css/source/components/_widget.scss
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/css/source/lib/_algolia.scss
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/css/source/lib/_slick.scss
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/fonts/hind-bold-webfont.woff
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/fonts/hind-bold-webfont.woff2
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/fonts/hind-light-webfont.woff
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/fonts/hind-light-webfont.woff2
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/fonts/hind-medium-webfont.woff
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/fonts/hind-medium-webfont.woff2
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/fonts/hind-regular-webfont.woff
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/fonts/hind-regular-webfont.woff2
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/fonts/hind-semibold-webfont.woff
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/fonts/hind-semibold-webfont.woff2
[16:37:03] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/arrow-bottom-bold.svg
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/arrow-bottom.svg
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/arrow-up.svg
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/broom.svg
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/calendar-month.svg
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/calendar-small-blue.svg
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/calendar-small.svg
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/calendar.svg
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/chat.svg
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/checklist.svg
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/clock.svg
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/coins.svg
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/cross-close.png
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/cross-close.svg
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/handshake.svg
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/left-arrow-without-dash.svg
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/left-arrow.svg
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/list.svg
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/logo.svg
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/logos/logo-aubade-big.png
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/logos/logo-aubade-big@2x.png
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/logos/logo-aubade.png
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/logos/logo-aubade@2x.png
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/logos/logo-coin-project.png
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/logos/logo-coin-project@2x.png
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/logos/logo-ebat.png
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/logos/logo-ebat@2x.png
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/logos/logo-facebook.svg
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/logos/logo-googleplus.svg
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/logos/logo-guide-artisan.png
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/logos/logo-guide-artisan@2x.png
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/logos/logo-le-materiel-electrique.png
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/logos/logo-le-materiel-electrique@2x.png
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/logos/logo-les-materiaux-big.png
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/logos/logo-les-materiaux-big@2x.png
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/logos/logo-les-materiaux.png
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/logos/logo-les-materiaux@2x.png
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/logos/logo-materiel-electrique-big.png
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/logos/logo-materiel-electrique-big@2x.png
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/logos/logo-pinterest.svg
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/logos/logo-share.svg
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/logos/logo-project-text.png
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/logos/logo-project-text@2x.png
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/logos/logo-twitter.svg
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/logos/mini-logo-aubade.png
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/logos/mini-logo-aubade@2x.png
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/logos/mini-logo-ebat.png
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/logos/mini-logo-ebat@2x.png
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/logos/mini-logo-ga-big.png
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/logos/mini-logo-ga.png
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/logos/mini-logo-le-materiel-electrique.png
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/logos/mini-logo-le-materiel-electrique@2x.png
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/logos/mini-logo-les-materiaux.png
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/logos/mini-logo-les-materiaux@2x.png
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/logos/site-recrutement.png
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/logos/site-recrutement@2x.png
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/map-pin-empty.svg
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/map-pin-full-congress-blue.svg
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/map-pin-full-curious-blue.svg
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/map-pin-full.svg
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/map-pin.svg
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/menu.svg
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/mini-logo.png
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/mini-logo@2x.png
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/next.svg
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/previous.svg
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/right-arrow-without-dash.svg
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/right-arrow.svg
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/search.svg
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/settings.svg
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/images/street-view.svg
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/js/project/checkbox.js
[Browsersync] Reloading Browsers...
[16:37:04] Change detected. Theme: project_default File: var/view_preprocessed/frontools/frontend/Project/default/web/js/project/form.js
[Browsersync] Reloading Browsers...
[16:37:04] Change detected. Theme: project_default File: app/design/frontend/Project/default/web/css/source/components/_homepage.scss___jb_tmp___
[16:37:04] Resolving inheritance. Theme: project_default
[16:37:57] Theme: project_default main.css Compiled!
```

And the page has been refreshed.

**Comment by VincentMarmiesse** [comment]
Sometimes, after a modification on a SCSS file, it does nothing and then a few seconds later:

```
[17:04:29] Theme: project_default main.css Compiled!
[Browsersync] 1 file changed (main.css)
[17:04:35] Theme: project_default main.css Compiled!
[Browsersync] 1 file changed (main.css)
[17:04:35] Theme: project_default main.css Compiled!
[Browsersync] 1 file changed (main.css)
[17:04:35] Theme: project_default main.css Compiled!
[Browsersync] 1 file changed (main.css)
[17:04:35] Theme: project_default main.css Compiled!
[Browsersync] 1 file changed (main.css)
[17:04:35] Theme: project_default main.css Compiled!
[Browsersync] 1 file changed (main.css)
[17:04:35] Theme: project_default main.css Compiled!
[Browsersync] 1 file changed (main.css)
[17:04:35] Theme: project_default main.css Compiled!
[Browsersync] 1 file changed (main.css)
[17:04:35] Theme: project_default main.css Compiled!
[Browsersync] 1 file changed (main.css)
[17:04:35] Theme: project_default main.css Compiled!
[Browsersync] 1 file changed (main.css)
[17:04:35] Theme: project_default main.css Compiled!
[Browsersync] 1 file changed (main.css)
[17:04:35] Theme: project_default main.css Compiled!
[Browsersync] 1 file changed (main.css)
[17:04:35] Theme: project_default main.css Compiled!
[Browsersync] 1 file changed (main.css)
[17:04:35] Theme: project_default main.css Compiled!
[Browsersync] 1 file changed (main.css)
[17:04:35] Theme: project_default main.css Compiled!
[Browsersync] 1 file changed (main.css)
[17:04:35] Theme: project_default main.css Compiled!
[Browsersync] 1 file changed (main.css)
[17:04:35] Theme: project_default main.css Compiled!
[Browsersync] 1 file changed (main.css)
[17:04:35] Theme: project_default main.css Compiled!
[Browsersync] 1 file changed (main.css)
[17:04:35] Theme: project_default main.css Compiled!
[Browsersync] 1 file changed (main.css)
[17:04:35] Theme: project_default main.css Compiled!
[Browsersync] 1 file changed (main.css)
[17:04:35] Theme: project_default main.css Compiled!
[Browsersync] 1 file changed (main.css)
[17:04:35] Theme: project_default main.css Compiled!
[Browsersync] 1 file changed (main.css)
[17:04:35] Theme: project_default main.css Compiled!
[Browsersync] 1 file changed (main.css)
[17:04:35] Theme: project_default main.css Compiled!
[Browsersync] 1 file changed (main.css)
[17:04:35] Theme: project_default main.css Compiled!
[Browsersync] 1 file changed (main.css)
[17:04:35] Theme: project_default main.css Compiled!
[Browsersync] 1 file changed (main.css)
[17:04:35] Theme: project_default main.css Compiled!
[Browsersync] 1 file changed (main.css)
[17:04:35] Theme: project_default main.css Compiled!
[Browsersync] 1 file changed (main.css)
[17:04:35] Theme: project_default main.css Compiled!
[Browsersync] 1 file changed (main.css)
[17:04:35] Theme: project_default main.css Compiled!
[Browsersync] 1 file changed (main.css)
[17:04:35] Theme: project_default main.css Compiled!
[Browsersync] 1 file changed (main.css)
[17:04:35] Theme: project_default main.css Compiled!
[Browsersync] 1 file changed (main.css)
[17:04:35] Theme: project_default main.css Compiled!
[Browsersync] 1 file changed (main.css)
[17:04:35] Theme: project_default main.css Compiled!
[Browsersync] 1 file changed (main.css)
```

FYI, I only edited my SCSS file once.

**Comment by Igloczek** [comment]
Are you using some Vagrant / Docker setup?
By the previous version, you mean 1.4.x or older, right?

**Comment by VincentMarmiesse** [comment]
I am on Fedora 26, with Apache and PHP directly installed on my PC. I was previously using version 1.5.5 on a Mac with Parallels Desktop and a Ubuntu VM.

**Comment by janmannl** [comment]
I have the same issue, it seems that it generates the css files a couple of hundred times, as if it is looped. This process takes a long time and refreshing the page at that point is impossible, since the gulp thread is "clogged".

**Comment by Igloczek** [comment]
Could you try to track down the last working well version to simplify the debugging process? I'm extremely busy now, but want to help you somehow in same time :/

**Comment by danslo** [comment]
I ran into the same issue, and did a git bisect to find the offending commit:

```
cb1375d8308b204eb2e2ab0831dc51ba76f18843 is the first bad commit
commit cb1375d8308b204eb2e2ab0831dc51ba76f18843
Author: Bartek Igielski <igloczek@gmail.com>
Date:   Sat Jun 17 20:40:02 2017 +0200

    Watch task refactor. Dropped support for lang overwrite and Windows
```

Before this commit, all is fine and dandy.

**Comment by Igloczek** [comment]
@danslo It's unfortunately not so helpful, b/c that's the full rewrite of watching logic and I'm not going to revert all of it, b/c it works waaaaay better than before (except few ppl, probably due to specific local env config, have some issues).

**Comment by danslo** [comment]
I'm not sure, I feel like I have a _very_ minimal config, with the only thing I changed is themes.json:

```json
{
  "blank": {
    "src": "vendor/snowdog/theme-blank-sass",
    "dest": "pub/static/frontend/Snowdog/blank",
    "locale": ["nl_NL"]
  },
  "kings": {
    "src": "app/design/frontend/Kings/default",
    "dest": "pub/static/frontend/Kings/default",
    "locale": ["nl_NL"],
    "parent": "blank"
  }
}
```

And then a very basic setup in app/design/frontend/Kings/default (though, not sure if it's relevant):

theme.xml:
```xml
<theme xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:Config/etc/theme.xsd">
    <title>Kings</title>
    <parent>Snowdog/blank</parent>
</theme>
```

registration.php:
```php
<?php

use Magento\Framework\Component\ComponentRegistrar;

ComponentRegistrar::register(
    ComponentRegistrar::THEME,
    'frontend/Kings/default',
    __DIR__
);
```

And then styles/styles.scss is a copy of snowdog theme, with single added line of ``@import 'kings.scss';``. 

And I'm just using ``gulp dev --theme kings``.

That's all I have, so I'm not sure about exotic/weird configuration issues. Maybe I'm missing something extremely obvious.

p.s. I have anohter issue where Browsersync is connected but it won't inject my new changes either, though, it didn't do so either in pre-1.5.0 versions so I'll view that as a separate issue (but perhaps related).

**Comment by danslo** [comment]
I'll try to debug further but I just wanted to check if I'm going about this the right way.

**Comment by Igloczek** [comment]
Do no use `.scss` in imports, just run `gulp sass-lint` and it will yell at you that it's wrong :). Every import needs to be `@import 'best/path/to/file` when the file is in `best/path/to/_file.scss`.

Any Magento stuff like `registration.php` or `theme.xml` doesn't matter for Frontools, it read only `themes.json`.

By exotic configurations, I mean OS, partitions setup, virtualization, containers etc.

**Comment by danslo** [comment]
Sorry, the filename.scss was a quick typo from the top of my head. ``gulp styles`` works as expected, so it's definitely something related to the watcher code. I'll try to debug this some more with my minimal frontend tool knowledge :)

**Comment by Igloczek** [comment]
Sass compilation works different than the watcher, that's why it's important to keep that imports in the way I mentioned, b/c tools are parsing content of all `.scss` files and search for imports. Using this data tools are creating the tree of dependent files to watch, so if you are using the not compatible style of import, the watcher will not work.

**Comment by danslo** [comment]
So the watch helper does something like this (added the logging myself):

```js
        // Disable watcher to not fire tons of events while solving inheritance
        plugins.util.log('Unwatching: ' + themeTempSrc);
        tempWatcher.unwatch(themeTempSrc);
```

Yet, when I gulp dev, we get the following when editing a single scss file:

```bash
[13:57:11] Unwatching: /home/daniel/Code/magento2-store-kings/var/view_preprocessed/frontools/frontend/Kings/default
[13:57:11] Change detected. Theme: kings File: var/view_preprocessed/frontools/frontend/Kings/default/styles/kings.scss
[13:57:11] Change detected. Theme: kings File: var/view_preprocessed/frontools/frontend/Kings/default/Magento_AdvancedCheckout/styles/_module.scss
[13:57:11] Change detected. Theme: kings File: var/view_preprocessed/frontools/frontend/Kings/default/Magento_AdvancedCheckout/styles/_widgets.scss
[...]
```

So I'm guessing something goes wrong with unwatching files in chokidar plugin.

**Comment by Igloczek** [comment]
Apart from possible chokidar issues - Why you have `kings.scss` if you are using this as a partial? It should be `_kings.scss`.

Also, the inheritance should be solved only when you are moving or adding files, is that true in your case?

**Comment by danslo** [comment]
>Apart from possible chokidar issues - Why you have kings.scss if you are using this as a partial? It should be _kings.scss.

Not sure why I did that, just wanted to have something very basic (and it worked with ``gulp styles``).

>Also, the inheritance should be solved only when you are moving or adding files, is that true in your case?

Nope, that's the thing. This kings.scss file contains only a  ``body { background-color: #123; }`` which I just change randomly to see if it updates on save. When I save this file, it create a huge list of ``Change detected.`` for that view_preprocessed folder.

**Comment by danslo** [comment]
Looks like we're on the right track:

```bash
[14:30:44] Watcher initialized! Theme: kings and dependencies...
add /home/daniel/Code/magento2-store-kings/app/design/frontend/Kings/default/styles/_kings.scss___jb_tmp___
change /home/daniel/Code/magento2-store-kings/app/design/frontend/Kings/default/styles/_kings.scss
[14:30:52] Change detected. Theme: kings File: var/view_preprocessed/frontools/frontend/Kings/default/styles/_kings.scss
[14:30:52] Change detected. Theme: kings File: app/design/frontend/Kings/default/styles/_kings.scss___jb_tmp___
[14:30:52] Resolving inheritance. Theme: kings
unlink /home/daniel/Code/magento2-store-kings/app/design/frontend/Kings/default/styles/_kings.scss___jb_tmp___
[14:30:52] Change detected. Theme: kings File: var/view_preprocessed/frontools/frontend/Kings/default/Magento_AdvancedCheckout/styles/_module.scss
[etc]
```

Then googling for this ``___jb_tmp___`` file:
https://github.com/paulmillr/chokidar/issues/493

It's some kind of temporary file added by PHPStorm IDE. I disabled 'safe write' feature as described here:
https://stackoverflow.com/questions/23271895/temporary-jb-old-file-in-phpstorm-and-webstorm-causing-errors

And all was good in the world!

**Comment by VincentMarmiesse** [comment]
Hi guys,

Just for information, I'm using PHPStorm too.

Thanks for your investigation!

**Comment by danslo** [comment]
At least I can edit scss files now, but adding/remove files will cause gulp task to go 500% CPU usage. After maybe 1 or 2 minutes (on a latest gen i7), will get something like:

```
[...]
[14:53:48] Theme: kings File: _utilities.scss - SASS Lint finished.
[14:53:48] Theme: kings File: _actions-toolbar.scss - SASS Lint finished.
[14:53:48] Theme: kings File: _breadcrumbs.scss - SASS Lint finished.
[14:53:48] Theme: kings File: _buttons.scss - SASS Lint finished.
[14:53:48] Theme: kings File: _dropdowns.scss - SASS Lint finished.
[14:53:48] Theme: kings File: _colors.scss - SASS Lint finished.
[14:53:48] Theme: kings File: _components.scss - SASS Lint finished.
[14:53:48] Theme: kings File: _email.scss - SASS Lint finished.
[14:53:48] Theme: kings File: _global.scss - SASS Lint finished.
[14:53:48] Theme: kings File: _forms.scss - SASS Lint finished.
[14:53:48] Theme: kings File: _icons.scss - SASS Lint finished.
[14:53:48] Theme: kings File: _layout.scss - SASS Lint finished.
[14:53:48] Theme: kings File: _loaders.scss - SASS Lint finished.
[14:53:48] Theme: kings File: _messages.scss - SASS Lint finished.
[14:53:48] Theme: kings File: _navigation.scss - SASS Lint finished.
[14:53:48] Theme: kings File: _rating.scss - SASS Lint finished.
[14:53:48] Theme: kings File: _pages.scss - SASS Lint finished.
[14:53:48] Theme: kings File: _popups.scss - SASS Lint finished.
[14:53:48] Theme: kings File: _responsive.scss - SASS Lint finished.
[14:53:48] Theme: kings File: _sections.scss - SASS Lint finished.
[14:53:48] Theme: kings File: _structure.scss - SASS Lint finished.
[14:53:48] Theme: kings File: _tables.scss - SASS Lint finished.
[14:53:48] Theme: kings File: _tooltips.scss - SASS Lint finished.
[14:53:48] Theme: kings File: _typography.scss - SASS Lint finished.
[14:53:48] Theme: kings File: _lib.scss - SASS Lint finished.
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings print.css Compiled!
[Browsersync] 1 file changed (print.css)
[14:55:11] Theme: kings print.css Compiled!
[Browsersync] 1 file changed (print.css)
[14:55:11] Theme: kings print.css Compiled!
[Browsersync] 1 file changed (print.css)
[14:55:11] Theme: kings print.css Compiled!
[Browsersync] 1 file changed (print.css)
[14:55:11] Theme: kings print.css Compiled!
[Browsersync] 1 file changed (print.css)
[14:55:11] Theme: kings print.css Compiled!
[Browsersync] 1 file changed (print.css)
[14:55:11] Theme: kings print.css Compiled!
[Browsersync] 1 file changed (print.css)
[14:55:11] Theme: kings print.css Compiled!
[Browsersync] 1 file changed (print.css)
[14:55:11] Theme: kings print.css Compiled!
[Browsersync] 1 file changed (print.css)
[14:55:11] Theme: kings print.css Compiled!
[Browsersync] 1 file changed (print.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings print.css Compiled!
[Browsersync] 1 file changed (print.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings print.css Compiled!
[Browsersync] 1 file changed (print.css)
[14:55:11] Theme: kings print.css Compiled!
[Browsersync] 1 file changed (print.css)
[14:55:11] Theme: kings print.css Compiled!
[Browsersync] 1 file changed (print.css)
[14:55:11] Theme: kings print.css Compiled!
[Browsersync] 1 file changed (print.css)
[14:55:11] Theme: kings print.css Compiled!
[Browsersync] 1 file changed (print.css)
[14:55:11] Theme: kings print.css Compiled!
[Browsersync] 1 file changed (print.css)
[14:55:11] Theme: kings print.css Compiled!
[Browsersync] 1 file changed (print.css)
[14:55:11] Theme: kings print.css Compiled!
[Browsersync] 1 file changed (print.css)
[14:55:11] Theme: kings print.css Compiled!
[Browsersync] 1 file changed (print.css)
[14:55:11] Theme: kings print.css Compiled!
[Browsersync] 1 file changed (print.css)
[14:55:11] Theme: kings print.css Compiled!
[Browsersync] 1 file changed (print.css)
[14:55:11] Theme: kings print.css Compiled!
[Browsersync] 1 file changed (print.css)
[14:55:11] Theme: kings print.css Compiled!
[Browsersync] 1 file changed (print.css)
[14:55:11] Theme: kings print.css Compiled!
[Browsersync] 1 file changed (print.css)
[14:55:11] Theme: kings print.css Compiled!
[Browsersync] 1 file changed (print.css)
[14:55:11] Theme: kings print.css Compiled!
[Browsersync] 1 file changed (print.css)
[14:55:11] Theme: kings print.css Compiled!
[Browsersync] 1 file changed (print.css)
[14:55:11] Theme: kings print.css Compiled!
[Browsersync] 1 file changed (print.css)
[14:55:11] Theme: kings print.css Compiled!
[Browsersync] 1 file changed (print.css)
[14:55:11] Theme: kings print.css Compiled!
[Browsersync] 1 file changed (print.css)
[14:55:11] Theme: kings print.css Compiled!
[Browsersync] 1 file changed (print.css)
[14:55:11] Theme: kings print.css Compiled!
[Browsersync] 1 file changed (print.css)
[14:55:11] Theme: kings print.css Compiled!
[Browsersync] 1 file changed (print.css)
[14:55:11] Theme: kings print.css Compiled!
[Browsersync] 1 file changed (print.css)
[14:55:11] Theme: kings print.css Compiled!
[Browsersync] 1 file changed (print.css)
[14:55:11] Theme: kings print.css Compiled!
[Browsersync] 1 file changed (print.css)
[14:55:11] Theme: kings print.css Compiled!
[Browsersync] 1 file changed (print.css)
[14:55:11] Theme: kings print.css Compiled!
[Browsersync] 1 file changed (print.css)
[14:55:11] Theme: kings print.css Compiled!
[Browsersync] 1 file changed (print.css)
[14:55:11] Theme: kings print.css Compiled!
[Browsersync] 1 file changed (print.css)
[14:55:11] Theme: kings print.css Compiled!
[Browsersync] 1 file changed (print.css)
[14:55:11] Theme: kings print.css Compiled!
[Browsersync] 1 file changed (print.css)
[14:55:11] Theme: kings print.css Compiled!
[Browsersync] 1 file changed (print.css)
[14:55:11] Theme: kings print.css Compiled!
[Browsersync] 1 file changed (print.css)
[14:55:11] Theme: kings print.css Compiled!
[Browsersync] 1 file changed (print.css)
[14:55:11] Theme: kings print.css Compiled!
[Browsersync] 1 file changed (print.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
[14:55:11] Theme: kings styles.css Compiled!
[Browsersync] 1 file changed (styles.css)
```

Even though I only added a single file in my theme, once, and didn't even import it anywhere. Not sure why it tried to compile styles.css about 230 times... More investigation is needed :-P

**Comment by danslo** [comment]
OK, so to me it seems like some sort of timing issue:

1) reinitialize function calls chokidar unwatch() to disable events in temporary folder while solving inheritance
2) inheritance solver then does a bunch of _synchronous_ calls to remove/create symlinks.
3) temporary folder is added again to watcher
4) only after all this has happened, we start to get ``change`` events for the temp watcher.

**Comment by Igloczek** [comment]
Could you try to change https://github.com/SnowdogApps/magento2-frontools/blob/master/task/watch.js#L89 to `tempWatcher.close();`, this should definitely kill all listeners.

**Comment by danslo** [comment]
Yeah it kills the listeners, and therefore never finds new files in view_preprocessed and therefore never compiles styles.css anymore. 

I think _unwatch_ is the proper way to go, but there's a synchronicity issue.

**Comment by danslo** [comment]
To better illustrate:

```
+++ Unwatching: /home/daniel/Code/magento2-store-kings/var/view_preprocessed/frontools/frontend/Kings/default
[...]
+++ Creating symlink in: /home/daniel/Code/magento2-store-kings/var/view_preprocessed/frontools/frontend/Kings/default/Magento_AdvancedCheckout/styles/_module.scss
+++ Creating symlink in: /home/daniel/Code/magento2-store-kings/var/view_preprocessed/frontools/frontend/Kings/default/Magento_AdvancedCheckout/styles/_widgets.scss
[more symlinks created]
+++ Watching: /home/daniel/Code/magento2-store-kings/var/view_preprocessed/frontools/frontend/Kings/default
[...]
+++ Got change event for path: /home/daniel/Code/magento2-store-kings/var/view_preprocessed/frontools/frontend/Kings/default/Magento_AdvancedCheckout/styles/_module.scss
+++ Got change event for path: /home/daniel/Code/magento2-store-kings/var/view_preprocessed/frontools/frontend/Kings/default/Magento_AdvancedCheckout/styles/_widgets.scss
[more change events]
```
And thus triggering recompilation of styles.scss many times. Not really sure how to solve it though.

**Comment by Igloczek** [comment]
Let's check two things:
1. How many times `tempWatcher.emit(change', file);` is fired and for what files. It should be fired once if you changed one file.
2. Check the content of `sassDependecyTree`, length of the object keys should be equal to number of "root" `scss` files (in your case probably 1, because you have just `styles.scss`)

For me the order of executions seems to be correct, one think that bothers me is if the inheritance task really finished its job, before adding back files to the watcher, but I'm not sure if it's possible, b/c all things are synchronous and inheritance resolver is returning a promise.

**Comment by danslo** [comment]
So, the only way I got this properly working is by closing the watcher, then just recreating it after solving inheritance.

```diff
diff --git task/watch.js task/watch.js
index 42f251e..ea256ab 100755
--- task/watch.js
+++ task/watch.js
@@ -42,12 +42,12 @@ module.exports = function(resolve) { // eslint-disable-line func-names
     };
 
     // Initialize watchers
-    const tempWatcher = plugins.chokidar.watch(themeTempSrc, watcherConfig), // eslint-disable-line one-var
-          srcWatcher = plugins.chokidar.watch(themeSrc, watcherConfig),
+    const srcWatcher = plugins.chokidar.watch(themeSrc, watcherConfig),
           destWatcher = plugins.chokidar.watch(themeDest, watcherConfig);
 
     let reinitTimeout = false,
-        sassDependecyTree = {};
+        sassDependecyTree = {},
+        tempWatcher = getTempWatcher();
 
     function generateSassDependencyTree() {
       // Cleanup tree
@@ -63,6 +63,66 @@ module.exports = function(resolve) { // eslint-disable-line func-names
       });
     }
 
+    function getTempWatcher() {
+      var watcher = plugins.chokidar.watch(themeTempSrc, watcherConfig);
+
+      watcher.on('ready', () => {
+        plugins.util.log(
+          plugins.util.colors.yellow('Watcher initialized!') + ' ' +
+          plugins.util.colors.green('Theme:') + ' ' +
+          plugins.util.colors.blue(name) + ' ' +
+          plugins.util.colors.green('and dependencies...')
+        );
+      });
+
+      // Events handling
+      watcher.on('change', path => {
+        // Print message to know what's going on
+        plugins.util.log(
+          plugins.util.colors.yellow('Change detected.') + ' ' +
+          plugins.util.colors.green('Theme:') + ' ' +
+          plugins.util.colors.blue(name) + ' ' +
+          plugins.util.colors.green('File:') + ' ' +
+          plugins.util.colors.blue(plugins.path.relative(config.projectPath, path))
+        );
+
+        // SASS Lint
+        if (!plugins.util.env.disableLinting) {
+          if (plugins.path.extname(path) === '.scss') {
+            plugins.helper.sassLint(gulp, plugins, config, name, path);
+          }
+        }
+
+        // SASS Compilation
+        if (plugins.path.extname(path) === '.scss') {
+          Object.keys(sassDependecyTree).forEach(file => {
+            if (sassDependecyTree[file].includes(path)) {
+              plugins.helper.sass(gulp, plugins, config, name, file);
+            }
+          });
+        }
+
+        // Babel
+        if (plugins.path.basename(path).includes('.babel.js')) {
+          plugins.helper.babel(gulp, plugins, config, name, path);
+        }
+
+        // SVG Sprite
+        if (plugins.path.extname(path) === '.svg') {
+          plugins.helper.svg(gulp, plugins, config, name);
+        }
+
+        // Files that require reload after save
+        if (['.html', '.phtml', '.xml', '.csv', '.js'].some(ext => {
+          return plugins.path.extname(path) === ext;
+        })) {
+          plugins.browserSync.reload();
+        }
+      });
+
+      return watcher;
+    }
+
     generateSassDependencyTree();
 
     function reinitialize(path) {
@@ -86,7 +146,7 @@ module.exports = function(resolve) { // eslint-disable-line func-names
         );
 
         // Disable watcher to not fire tons of events while solving inheritance
-        tempWatcher.unwatch(themeTempSrc);
+        tempWatcher.close();
 
         // Run inheritance resolver just for one theme without parent(s)
         plugins.helper.inheritance(plugins, config, name, false).then(() => {
@@ -94,7 +154,7 @@ module.exports = function(resolve) { // eslint-disable-line func-names
           generateSassDependencyTree();
 
           // Add all files to watch again after solving inheritance
-          tempWatcher.add(themeTempSrc);
+          tempWatcher = getTempWatcher();
 
           // Emit event on added / moved / renamed / deleted file to trigger regualr pipeline
           plugins.globby.sync(themeTempSrc + '/**/' + plugins.path.basename(path))
@@ -112,61 +172,6 @@ module.exports = function(resolve) { // eslint-disable-line func-names
       .on('unlink', reinitialize)
       .on('unlinkDir', reinitialize);
 
-    // print msg when temp dir watcher is initialized
-    tempWatcher.on('ready', () => {
-      plugins.util.log(
-        plugins.util.colors.yellow('Watcher initialized!') + ' ' +
-        plugins.util.colors.green('Theme:') + ' ' +
-        plugins.util.colors.blue(name) + ' ' +
-        plugins.util.colors.green('and dependencies...')
-      );
-    });
-
-    // Events handling
-    tempWatcher.on('change', path => {
-      // Print message to know what's going on
-      plugins.util.log(
-        plugins.util.colors.yellow('Change detected.') + ' ' +
-        plugins.util.colors.green('Theme:') + ' ' +
-        plugins.util.colors.blue(name) + ' ' +
-        plugins.util.colors.green('File:') + ' ' +
-        plugins.util.colors.blue(plugins.path.relative(config.projectPath, path))
-      );
-
-      // SASS Lint
-      if (!plugins.util.env.disableLinting) {
-        if (plugins.path.extname(path) === '.scss') {
-          plugins.helper.sassLint(gulp, plugins, config, name, path);
-        }
-      }
-
-      // SASS Compilation
-      if (plugins.path.extname(path) === '.scss') {
-        Object.keys(sassDependecyTree).forEach(file => {
-          if (sassDependecyTree[file].includes(path)) {
-            plugins.helper.sass(gulp, plugins, config, name, file);
-          }
-        });
-      }
-
-      // Babel
-      if (plugins.path.basename(path).includes('.babel.js')) {
-        plugins.helper.babel(gulp, plugins, config, name, path);
-      }
-
-      // SVG Sprite
-      if (plugins.path.extname(path) === '.svg') {
-        plugins.helper.svg(gulp, plugins, config, name);
-      }
-
-      // Files that require reload after save
-      if (['.html', '.phtml', '.xml', '.csv', '.js'].some(ext => {
-        return plugins.path.extname(path) === ext;
-      })) {
-        plugins.browserSync.reload();
-      }
-    });
-
     destWatcher.on('change', path => {
       // CSS Lint
       if (!plugins.util.env.disableLinting) {
```

This feels _dirty_ to me, but hey, it works.

**Comment by danslo** [comment]
>For me the order of executions seems to be correct, one think that bothers me is if the inheritance task really finished its job, before adding back files to the watcher, but I'm not sure if it's possible, b/c all things are synchronous and inheritance resolver is returning a promise.

I think the problem here is that we're confusing what node-fs-extra is actually doing when it talks about doing things synchronously. In this case [ensureLinkSync](https://github.com/jprichardson/node-fs-extra/blob/HEAD/docs/ensureLink-sync.md) will just block until it has written the symlink to harddisk. No guarantees made about inotify events or anything.

**Comment by Igloczek** [comment]
I'm not sure if I understand this topic good enough (first time hear about `inotify`), but as long the thread block is long enough and inheritance resolver work synchronously, we are good, because listening to any events while resolving inheritance isn't the goal.

In my understanding and looking at the proposed solution, the problem seems to be on chokidar `unwatch` method, because, using some more force, we get exactly what this method should do - unwatched directory.

**Comment by danslo** [comment]
To further illustrate why I think this is a timing / order of execution issue, when I enable polling in watcher.json, the problem goes away! I guess the poll window gives just enough time to resolve inheritance without immediately triggering change events.

**Comment by danslo** [comment]
Thanks for your patience and this awesome project, by the way!

**Comment by Igloczek** [comment]
Could you try to add some logs to function responsible for creating symlinks within inheritance resolver and to the code that should run after it finishes in watcher, to check if there is really a chance that some symlinks are created after the task was finished?

IMO even if there will be some "lag" in emiting low-level events it will be realted to few files, not 200 of them.

**Comment by Igloczek** [comment]
Thanks for the investigation Daniel!

I tried to achieve similar effects on my machine and rapid changes sometimes trigger unwanted behavior.

After spending a lot of time trying to make the whole process more organized by queuing tasks that are not meant to run in parallel, like solving inheritance and... nothing changed, it was even worse.

At all, I found some ways to improve the code and I released it as version 1.5.11

**Comment by danslo** [comment]
I tried 1.5.11 but with polling disabled I get the same issue. Maybe one of the reasons that I can reproduce it consistently might be that I'm running everything natively (no containers / VMs / network drives) on linux on a rather fast (M.2) SSD. 

In any case, thanks for your patience, I'm already happy I can work around the issue using polling.

**Comment by Igloczek** [comment]
Closing this, since I don't see anything that can in the code to fix reported issues for everyone.

For now the fix is to set `"usePolling": true` in `config/watcher.json`
